src/Servers/HttpSys/src/NativeInterop/DisconnectListener.cs

```diff
@@ -59,35 +56,39 @@ private ConnectionCancellation GetCreatedConnectionCancellation(ulong connection
 
         private unsafe CancellationToken CreateDisconnectToken(ulong connectionId)
         {
-            _logger.LogDebug(LoggerEventIds.RegisterDisconnectListener, "CreateDisconnectToken; Registering connection for disconnect for connection ID: {0}" , connectionId);
+            _logger.LogDebug(LoggerEventIds.RegisterDisconnectListener, "CreateDisconnectToken; Registering connection for disconnect for connection ID: {0}", connectionId);
 
             // Create a nativeOverlapped callback so we can register for disconnect callback
             var cts = new CancellationTokenSource();
             var returnToken = cts.Token;
 
-            SafeNativeOverlapped nativeOverlapped = null;
-            var boundHandle = _requestQueue.BoundHandle;
-            nativeOverlapped = new SafeNativeOverlapped(boundHandle, boundHandle.AllocateNativeOverlapped(
-                (errorCode, numBytes, overlappedPtr) =>
+            var overlapped = new Overlapped
+            {
+                OffsetHigh = 0,
+                OffsetLow = 0
+            };
+
+            // We're not using boundHandle.AllocateNativeOverlapped here because we want to avoid capturing the ExecutionContext (see https://github.com/dotnet/runtime/issues/42549)
+            // Instead, we're going to use lower level APIs to get access to UnsafePack (which avoids the capture)
+            var nativeOverlapped = overlapped.UnsafePack((errorCode, numBytes, pOverlapped) =>
```

Comment:
Can we add a version of SafeNativeOverlapped that doesn't take a ThreadPoolBoundHandle and instead just calls Overlapped.Free? It would feel safer given this was managed by a SafeHandle previously and the lifetime isn't super obvious.

Reply:
Maybe in the backport but I don't think in 6.0. I'm actually not convinced we should be using safe handle here anyways.

Reply:
I also feel pretty confident about this change mostly because we don't call dispose on the safe handle outside of this code.